libs/dyn/jsonloader/locations.go

```diff
@@ -0,0 +1,44 @@
+package jsonloader
+
+import (
+	"sort"
+
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type LineOffset struct {
+	Line  int
+	Start int64
+}
+
+// buildLineOffsets scans the input data and records the starting byte offset of each line.
+func BuildLineOffsets(data []byte) []LineOffset {
```

Comment:
This can return an offsets object and then the function below can be a member function.

The slice itself doesn't need to be changed once it is built.